File: javascript/leetcode/1437. Check If All 1's Are at Least Length K Places Away.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var kLengthApart = function (nums, k) {
  let last = -1;
  let first = -1;
  const sz = nums.length;

  for (let i = 0; i < sz; ++i) {
    if (nums[i] === 1) {
      if (first < 0) first = i;
      else if (i - last - 1 < k) return false;

      last = i;
    }
  }

  return true;
};

File: c++/leetcode/1437. Check If All 1's Are at Least Length K Places Away.cpp
class Solution
{
public:
  bool kLengthApart(vector<int> &nums, int k)
  {
    const int sz = nums.size();
    int first = -1;
    int last = -1;

    for (int i = 0; i < sz; ++i)
    {
      if (nums[i] == 1)
      {
        if (first < 0)
          first = i;
        else if ((i - last) - 1 < k)
          return false;

        last = i;
      }
    }

    return true;
  }
};